### DISCUSSION TITLE
Is it possible to have a ContextMenu with background transparent blue?

### DISCUSSION THREAD

**Discussion by Ehrode** [AvaloniaUI/Avalonia#12640] [Q&A] [answered]
Hello,
Everything is in the question, is it possible to achieve this effect: 
![image](https://github.com/AvaloniaUI/Avalonia/assets/40527875/473b2942-5b62-450d-b40b-bc10612a6037)

but for a control like a context menu?

I tried to play around with ExperimentalAcrylicBorder as Control Template but their is no way to get any transparency on my context menu.

Ty

**Comment by timunie** [comment]
Have a look what FluentAvalonia does to get transparency: https://github.com/amwx/FluentAvalonia/blob/4d847a8403b914d558baa79351a7866a02ec0b3e/src/FluentAvalonia/Styling/ControlThemes/BasicControls/PopupRootStyles.axaml#L5

**Reply by Ehrode** [reply]
Yeah I might be able to achieve it not by templating a new control based on a window or a popup that have a TransparencyLevelHint property. But my question is more about possibility of styling the ContextMenu control (which doesn't have the TransparencyLevelHint) to have this effect without having to make a new control from scratch.

**Reply by timunie** [reply, marked as answer]
Have you tried to add this style? PopupRoot is part of the control, so it should help. Also lookup the original styles for other ContextMenu related styles

**Reply by Ehrode** [reply]
I'm not sure to understand your response. Do you mean PopupRoot is part of context menu? When I lookup the fluent theme of context menu I can't see any reference to PopupRoot: 
https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Themes.Fluent/Controls/ContextMenu.xaml

So styling PopupRoot doesn't affect the ContextMenu. Do you mean I should replace my context menu template by a PopupRoot?

Edit; Indeed you are right, changing the style of PopupRoot affects my context menu!
However, by doing so I get exactly the same result as directly overriding the context menu template:
The acrylic texture is present but without any transparency.

<Setter Property="Background" Value="{x:Null}" />
<Setter Property="TransparencyLevelHint" Value="Transparent" />

I have those two lines and I added the ExperimentalAcrylicBorder to the PopupRoot Panel

**Reply by Ehrode** [reply]
By tweaking values of acrylic border I finally get it to kinda work;
![image](https://github.com/AvaloniaUI/Avalonia/assets/40527875/081e4499-dd11-4eff-afec-41d1c443d243)
Now I get this weird behavior when it works outside my main window but not over the window itself.
Thank you for your answers it helped a lot, I'll continue to play around and see if I can get it to work how I'd like!
